src/main/java/io/github/debuggyteam/tablesaw/TableSawResourceLoader.java

```diff
@@ -0,0 +1,239 @@
+package io.github.debuggyteam.tablesaw;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import org.jetbrains.annotations.NotNull;
+import org.quiltmc.loader.api.QuiltLoader;
+import org.quiltmc.qsl.resource.loader.api.reloader.SimpleSynchronousResourceReloader;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+
+import io.github.debuggyteam.tablesaw.api.TableSawAPI;
+import io.github.debuggyteam.tablesaw.api.TableSawCompat;
+import io.github.debuggyteam.tablesaw.api.TableSawRecipe;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.StringNbtReader;
+import net.minecraft.resource.Resource;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class TableSawResourceLoader implements SimpleSynchronousResourceReloader {
+	public static final Identifier ID = new Identifier(TableSaw.MODID, "recipe_loader");
+	
+	@Override
+	public @NotNull Identifier getQuiltId() {
+		return ID;
+	}
+	
+	@Override
+	public void reload(ResourceManager manager) {
+		//We start fresh at every datapack reload
+		TableSawRecipes.serverInstance().clearAllRecipes();
+		
+		//TODO: Heuristics
+		
+		//Create the API implementation
+		TableSawAPI api = new TableSawAPI() {
+			@Override
+			public void registerTableSawRecipe(TableSawRecipe recipe) {
+				TableSawRecipes.serverInstance().registerRecipe(recipe);
+				
+			}
+		};
+		
+		//Call all the API consumers
+		for(TableSawCompat compat : QuiltLoader.getEntrypoints(TableSaw.MODID, TableSawCompat.class)) {
+			compat.run(api);
+		}
+		
+		//Load data recipes, including our builtins
+		Map<Identifier, Resource> tableSawRecipes = manager.findResources("custom_recipes/tablesaw", (id)->id.getPath().endsWith(".json"));
+		for(Map.Entry<Identifier, Resource> resource : tableSawRecipes.entrySet()) {
+			
+			try {
+				String recipeString = new String(resource.getValue().open().readAllBytes(), StandardCharsets.UTF_8);
+				JsonElement elem = JsonParser.parseString(recipeString);
+				if (elem instanceof JsonArray array) {
+					for(JsonElement arrayItem : array) {
+						if (arrayItem instanceof JsonObject objItem) {
+							TableSawRecipe recipe = parseRecipe(objItem);
+							if (recipe != null) {
+								TableSawRecipes.serverInstance().registerRecipe(recipe);
+							}
+						}
+					}
+				} else if (elem instanceof JsonObject o) {
```

Comment:
`o` could become `object` so its a bit more descriptive. Same with `r` and `recipe` if doable.